### PR TITLE
Rank collapses plugin

### LINKED ISSUE #459: Rank should collapse the plugin, at least in some cases
When only 1 plugin is being displayed, and this is ranked by the user, the most likely desired behaviour is that the plugin is collapsed. This was the behaviour in the old report, btw :)

In general, we need to improve click-flow and reduce the number of clicks required to navigate the report, this is a small step in that direction

### PR DESCRIPTION
**Pull request by saganshul**
It fixes #459 

@7a @delta24 

Now on clicking the ranking button, that plugin gets collapse.

### CODE REVIEW

File: framework/interface/templates/plugin_report.html [hunk 1]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
-                                <label id="{{ pkey }}_5" class="btn {% if user_rank == 5 %}active btn-critical{% end %}" title="Critical"><input type="radio" id="5" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '5');"/><i class="fa fa-bomb"></i></label>
+                                <label id="{{ pkey }}_0" class="btn {% if user_rank == 0 %}active{% end %} rank" title="Passing" data-code="{{ test_code }}"><input type="radio" id="0" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '0');"/><i class="fa fa-thumbs-up"></i></label> 
+                                <label id="{{ pkey }}_1" class="btn {% if user_rank == 1 %}active btn-success{% end %} rank" title="Informational" data-code="{{ test_code }}"><input type="radio" id="1" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '1');"/><i class="fa fa-info-circle"></i></label>
+
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Remove extra empty line

File: framework/interface/templates/plugin_report.html [hunk 2]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
-                                <label id="{{ pkey }}_4" class="btn {% if user_rank == 4 %}active btn-danger{% end %}" title="High"><input type="radio" id="4" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '4');"/><i class="fa fa-bell"></i></label>
-                                <label id="{{ pkey }}_5" class="btn {% if user_rank == 5 %}active btn-critical{% end %}" title="Critical"><input type="radio" id="5" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '5');"/><i class="fa fa-bomb"></i></label>
+                                <label id="{{ pkey }}_0" class="btn {% if user_rank == 0 %}active{% end %} rank" title="Passing" data-code="{{ test_code }}"><input type="radio" id="0" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '0');"/><i class="fa fa-thumbs-up"></i></label>
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Remove extra space at the end of the line

File: framework/interface/templates/plugin_report.html [hunk 3]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
+                                <label id="{{ pkey }}_0" class="btn {% if user_rank == 0 %}active{% end %} rank" title="No ranking" data-code="{{ test_code }}"><input type="radio" id="0" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '0');"/><i class="fa fa-thumbs-up"></i></label>
+                                <label id="{{ pkey }}_1" class="btn rank {% if owtf_rank == 1 %}active btn-success" style="opacity: 0.7{% end %}" title="Informational" data-code="{{ test_code }}"><input type="radio" id="1" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '1');"/><i class="fa fa-info-circle"></i></label>
+                                <label id="{{ pkey }}_2" class="btn rank{% if owtf_rank == 2 %}active btn-info" style="opacity: 0.7{% end %}" title="Low" data-code="{{ test_code }}"><input type="radio" id="2" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '2');"/><i class="fa fa-exclamation-circle"></i></label>
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Missing space after `rank`

File: framework/interface/templates/plugin_report.html [hunk 4]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
+                                <label id="{{ pkey }}_3" class="btn rank{% if owtf_rank == 3 %}active btn-warning" style="opacity: 0.7{% end %}" title="Medium" data-code="{{ test_code }}"><input type="radio" id="3" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '3');"/><i class="fa fa-warning"></i></label>
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Missing space after `rank`

File: framework/interface/templates/plugin_report.html [hunk 5]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
+                                <label id="{{ pkey }}_4" class="btn rank{% if owtf_rank == 4 %}active btn-danger" style="opacity: 0.7{% end %}" title="High" data-code="{{ test_code }}"><input type="radio" id="4" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '4');"/><i class="fa fa-bell"></i></label>
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Missing space after `rank`

File: framework/interface/templates/plugin_report.html [hunk 6]
```diff
@@ -41,20 +41,21 @@ <h4 id="header_badge_{{ test_code }}"></h4>
...
+                                <label id="{{ pkey }}_5" class="btn {% if user_rank == 5 %}active btn-critical{% end %} rank" title="Critical" data-code="{{ test_code }}"><input type="radio" id="5" onchange="updatePluginInfoAndPatch('{{ pkey }}', '{{ poutput["plugin_group"] }}','{{ poutput["plugin_type"] }}','{{ test_code }}', '5');"/><i class="fa fa-bomb"></i></label>
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Keep consistency and move `rank` at the beginning. In fact, I think this one is more logical. Could you move all `rank` attributes at the end of the class items? Like you did in this line

File: framework/interface/templates/plugin_report.html [hunk 7]
```diff
@@ -385,4 +386,12 @@ <h4 id="header_badge_{{ test_code }}"></h4>
     updateTargetInfo();
     updateAllTestCasesInfo();
 });
+
+$(document).ready(function() {
+    $('.btn.rank').click(function() { 
+        var id = $(this).attr('data-code');
+        $('#' + id ).removeClass('in');
+        console.log(id);
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Remove debug output.

File: framework/interface/templates/plugin_report.html [hunk 8]
```diff
@@ -385,4 +386,12 @@ <h4 id="header_badge_{{ test_code }}"></h4>
     updateTargetInfo();
     updateAllTestCasesInfo();
 });
+
+$(document).ready(function() {
```
Comment (DePierre) [inline framework/interface/templates/plugin_report.html]: Is `$(document).ready()` necessary here?

Comment (delta24) [inline framework/interface/templates/plugin_report.html]: Yes, it will clash - load everything in one `$(document).ready()`.

**Comment by saganshul** [comment]
@DePierre  @7a PR updated please review.

File: framework/interface/templates/plugin_report.html [hunk 9]
```diff
@@ -381,8 +381,17 @@ <h4 id="header_badge_{{ test_code }}"></h4>
         $('#rankingTargetInfo').css("opacity", "0.7");
 }
 
+/* On Ranking the plugin, it collapses the plugin */
+function collapsePluginOnRank() {
+    $('.btn.rank').click(function() {
+        var id = $(this).attr('data-code');
+        $('#' + id ).removeClass('in');
```
Comment (delta24) [inline framework/interface/templates/plugin_report.html]: Remove the extra space after id

**Comment by saganshul** [comment]
@DePierre @7a @delta24 PR updated please review

**Comment by delta24** [comment]
Merged in 0368a6fd88d2f350c1eeb9d889b19e69ac73befe.

**Pull request closed without merging**